### ISSUE TITLE
docstrings on variables and type aliases are not treated as docstrings

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#21805]
### Summary

```py
# no docstring related errors are reported here:
asdf = 1
"""
a bunch of trailing spaces here that don't get removed by the formatter --->         
"""

# or here:
type Foo = int
"""
a bunch of trailing spaces here that don't get removed by the formatter --->         
"""

# errors are reported here though (D200, D212, D400, D401, D403, D415):
def foo():
    """
    these ones do --->              
    """
```

https://play.ruff.rs/ca57b564-80d5-42bd-a336-4e6ceb258320

### Version

0.14.8
